Cython/Compiler/Nodes.py

```diff
@@ -1678,6 +1679,8 @@ def align_argument_type(self, env, arg):
             return arg
         if other_type is None:
             error(type_node.pos, "Not a type")
+        elif other_type.is_fused and any(orig_type.same_as(t) for t in other_type.types):
+            arg.type = orig_type  # use specialized rather than fused type
```

Comment:
Isn't this just `pass`? Why reassign `.type` here?
I would rather avoid the assignment below in this case. Or use `pass` with a comment, in case there could be other special cases to handle in the future.

Reply:
Yes - I think this is just `pass`. I'll take it out.